interfaces/sourcegen/sourcegen/_dataclasses.py

```diff
@@ -59,12 +92,28 @@ def from_str(cls, arglist: str) -> 'ArgList':
             return cls([], spec)
         return cls([Param.from_str(arg) for arg in arglist.split(",")], spec)
 
+    @classmethod
+    def from_xml(cls, arglist: str) -> 'ArgList':
+        """Generate ArgList from XML string argument list."""
+        arglist, spec = cls._split_arglist(arglist)
+        if not arglist:
+            return cls([], spec)
+        return cls([Param.from_xml(arg) for arg in arglist.split(",")], spec)
+
+    def __len__(self):
+        return len(self.params)
+
     def __getitem__(self, k):
         return self.params[k]
 
     def __iter__(self) -> "Iterator[Param]":
         return iter(self.params)
 
+    def short_str(self) -> str:
+        """String representation of the argument list without parameter names."""
+        args = ', '.join([par.short_str() for par in self.params])
```

Comment:
I don't think you need to construct a list here:
```suggestion
        args = ', '.join(par.short_str() for par in self.params)
```